rclcpp/src/rclcpp/exceptions.cpp

```diff
@@ -73,6 +73,40 @@ throw_from_rcl_error(
   }
 }
 
+  std::exception_ptr
+  from_rcl_error(
+    rcl_ret_t ret,
+    const std::string & prefix,
+    const rcl_error_state_t * error_state,
+    void (* reset_error)())
+  {
+    if (RCL_RET_OK == ret) {
+      return std::make_exception_ptr(std::invalid_argument("ret is RCL_RET_OK"));
+    }
+    if (!error_state) {
+      error_state = rcl_get_error_state();
+    }
+    if (!error_state) {
+      return std::make_exception_ptr(std::runtime_error("rcl error state is not set"));
```

Comment:
Same comment about `throw std::runtime_error(...)` instead of returning the exception here since the exception didn't come from the value of `ret`.